tracing_actions/src/action_trace_subscriber.rs

```diff
@@ -237,7 +249,10 @@ impl<Sink: TraceSink + 'static, TSpanConstructor: SpanConstructor + 'static> Sub
             Some(mut closed_span) => {
                 closed_span.end();
                 log::trace!("Closed action span: {closed_span:?}");
-                self.span_sink.sink_trace(&mut closed_span);
+                if (self.should_record_span)(&mut closed_span) {
+                    log::trace!("Action span can be recorded: {closed_span:?}");
+                    self.span_sink.sink_trace(&mut closed_span);
+                }
```

Comment:
Hey, I see you're doing this way down at the sink edge. That's a fine place to do this work, however did you consider instead writing a wrapper sink? This is the way this kind of feature is intended to be supported:
```rust
struct SamplingSink<Downstream> {
    downstream: Downstream
}
impl<Downstream> TraceSink for SamplingSink<Downstream> where Downstream: TraceSink {
    fn sink_trace(&self, a_trace: &mut tracing_actions::ActionSpan) {
        if rand::random() {
            self.downstream.sink_trace(a_trace);
        }
    }
}
impl<Downstream> SamplingSink<Downstream> where Downstream: TraceSink {
    pub fn new(downstream: Downstream) -> Self {
        Self {
            downstream,
        }
    }
}
```

You see in this way tracing-actions always just calls your custom filtering sink, and your custom filtering sink can do whatever it needs to to decide whether _it_ wants to pass the span along to its sink. There's no more work to speak of that your application has to do in order to call your sink.

Doing this kind of feature the way I've suggested also gives you access to an `&self` in the decision whether to include the trace or not. You can do things with this like check for a particular attribute or consult an external rate limit selected by some measurement in the trace.

This way also keeps your generic types simpler. If you were to pass in a lambda for filtering, for example, you would find that you would have a difficult time storing the ActionTraceSubscriber if you had a reason to do so.

As-written, this would be a breaking change. Let's chat about whether the modular proxying sink pattern I mentioned above can accommodate your needs.